ChangeLog.md

```diff
@@ -4,21 +4,176 @@ Changelog
+    - Fixed calculation of some broken empire statistics
+    - Fixed a bug where ships could be ordered to invade but then not do so
+    - Fixed bug where a fleet would not block enemy supply propagation when it should have
+    - Accounted for production epsilon to avoid rounding errors in producible items list time estimates
+    - Fixed fighter hangar capacity being increased if any ship was owned by an empire with the appropriate tech, rather than just the ship's owner having it
+    - Fixed issue with ship designs being added with conflicting IDs when loading a game
+    - Fixed ships not getting the Interstellar Logistics speed bonus on the turn they were produced
+
+- Fixed some broken pedia links in stringtable entries
+- Fixed the sitrep "Last" button when the most-recent turn doesn't have any visible sitreps
+- Fixed parsed content checksum for ShipDesigns, which was looking up the design name in the stringtable
+- Fixed issue with options being reset / lost between executions within an app bundle
 
 #### Technical / Internal
 
-- C++14 is now supported and required to compile.
-- Removed OpenGL Utility (GLU) library dependency.
-- Provide FreeDesktop.org AppStream metainfo file.
+- Dependencies and Versions
+    - Updated to support and require Python 3.5+
+    - Raised C++ language version to C++14
+    - Raised minimum required version to build with CMake to 3.5
+    - Moved to compiling with Visual Studio 2017 on Windows
+    - Removed GLU library dependency
+    - Merged SDLGUI library into FreeOrion client
+
+- Scripting
+    - Optimized some condition internal evaluation and scripted implementation
+    - Optimized evaluation of effectsgroup scopes
+    - Optimized some effect execution
+    - Reworked hull and fuel regeneration part effects
+
+- Python AI
+    - Scripted AIs to scrap Gateways to the Void that they capture
+    - Reprioritized fuel tach research
+    - Reworked logic for responding to alliance requests
+    - Exposed option to append to a fleet route in the API
+    - Exposed productionLocation functions on ship parts and hulls to check if they can be produced at a location
+    - Wrapped calls to Python functions with error handlers that log exceptions
+    - Cached some results from calls to the API to improve performance
+    - Various other internal performance improvements
+    - Exposed empire stat records
+
+- FreeOrion Content Script Interface (FOCS)
+    - Added UsedInDesignID ValueRef to access the ID of the ship design in which a part or hull is used
+    - Added arrived-from system property value ref for fleets
+    - Added ShipDesignCost ComplexValueRef
+    - Made some ValueRefs return default values rather than throwing errors when invalid properties are evaluated
+    - Added missing empire ID handling in several conditions
+    - Replaced OwnerTradeStockpile with EmpireStockpile (that refers to industry stockpiled)
+    - Added NoDefaultFuelEffect flag to hulls
```

Comment:
Added NoDefaultSpeedEffect, NoDefaultFuelEffect, NoDefaultStealthEffect, and NoDefaultStructureEffect flag to hulls. Non-monster hulls now use NoDefaultFuelEffect to provide fuel efficiency